echopype/consolidate/api.py

```diff
@@ -55,84 +60,123 @@ def swap_dims_channel_frequency(ds: Union[xr.Dataset, str, pathlib.Path]) -> xr.
-    #         )
-
-    # # If not vertical needs to have tilt
-    # if not vertical:
-    #     if tilt is None:
-    #         if "tilt" in ds:
-    #             tilt = ds["tilt"]
-    #         else:
-    #             raise ValueError(
-    #                 "tilt not found in dataset and needs to be supplied by the user. "
-    #                 "Required when vertical=False"
-    #             )
-    # else:
-    #     tilt = 0
+    # Raise `NotImplementedError` if `use_platform_angles` and `use_beam_angles` are
+    # both true.
+    if use_platform_angles and use_beam_angles:
+        raise NotImplementedError(
+            "Computing depth with both platform and beam angles is not implemented yet."
+        )
 
-    ds = open_source(ds, "dataset", {})
-    # Multiplication factor depending on if transducers are pointing downward
-    mult = 1 if downward else -1
+    if echodata:
+        # Open Echodata
+        echodata = open_source(echodata, "echodata", {})
 
-    # Compute depth
-    ds["depth"] = mult * ds["echo_range"] * np.cos(tilt / 180 * np.pi) + depth_offset
-    ds["depth"].attrs = {"long_name": "Depth", "standard_name": "depth", "units": "m"}
+        # Grab sonar model
+        sonar_model = echodata["Sonar"].attrs["sonar_model"]
+
+        # Raise value error if sonar model is supported for `use_platform/beam_...` arguments
+        if sonar_model not in ["EK60", "EK80"] and (
+            use_platform_vertical_offsets or use_platform_angles or use_beam_angles
+        ):
+            raise NotImplementedError(
+                f"`use_platform/beam_...` not implemented yet for `{sonar_model}`."
+            )
+
+    # Compute transducer depth:
+    if use_platform_vertical_offsets and sonar_model in ["EK60", "EK80"]:
+        # Compute transducer depth in EK systems using platform vertical offset data
+        transducer_depth = ek_use_platform_vertical_offsets(echodata["Platform"], ds["ping_time"])
+    else:
+        # Compute transducer depth from user input depth offset
+        transducer_depth = depth_offset
+
+    # Compute echo range scaling:
+    if use_platform_angles and sonar_model in ["EK60", "EK80"]:
+        # Compute echo range scaling in EK systems using platform angle data
+        echo_range_scaling = ek_use_platform_angles(echodata["Platform"], ds["ping_time"])
+    elif use_beam_angles and sonar_model in ["EK60", "EK80"]:
+        # Identify beam group name by checking channel values of `ds`
+        if echodata["Sonar/Beam_group1"]["channel"].equals(ds["channel"]):
+            beam_group_name = "Beam_group1"
+        else:
+            beam_group_name = "Beam_group2"
+
+        # Compute echo range scaling in EK systems using beam angle data
+        echo_range_scaling = ek_use_beam_angles(echodata[f"Sonar/{beam_group_name}"], ds["channel"])
+    else:
+        # Compute echo range scaling from user input tilt
+        echo_range_scaling = np.cos(np.deg2rad(tilt))
```

Comment:
So a few thoughts:
- I think you can remove those conditions since you've checked `sonar_model` being one of EK60/EK80.
- Since you already checked the values of `tilt` and `use_platform_angles` and `use_beam_angles` and issue error or warnings, I think the structure could be simplified to be just `if` and no `else`, which makes the code easier to read.

With `depth_offset` and `tilt` taking precedence over the vertical offset and angles in the EchoData object, I think the logic here needs some rework. I'd suggest something like below as a clean decision tree structure:
```python
if depth_offset:
  # use depth offset since it takes precedence
else:
  # use vertical offset in EchoData if that's what the user wants and sonar_model is EK60/80
  if sonar_model in ["EK60", "EK80"]:
    if use_platform_angles:
      ...
    if use_beam_angles:
      ...
  else:
    ...

if tilt:
  # use tilt since it takes precedence
else:
  # use angles in EchoData if that's what the user wants and sonar_model is EK60/80
  if sonar_model in ["EK60", "EK80"]:
    ...
  else:
    ...

```
I've not tested the above so please check the tree is in good order.

Reply:
I agree I'll make this change 👍